sdk/azcore/tracing/constants.go

```diff
@@ -0,0 +1,41 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package tracing
+
+// SpanKind represents the role of a Span inside a Trace. Often, this defines how a Span will be processed and visualized by various backends.
+type SpanKind int
+
+const (
+	// SpanKindInternal indicates the span represents an interal operation within an application.  This is the default value.
```

Comment:
```suggestion
	// SpanKindInternal indicates the span represents an internal operation within an application.  This is the default value.
```
And should the value be 0, or does that indicate no setting?

Reply:
my 2 cents: internal should be a default kind at least for OTel implementation - OTel requires kind to be set.

Reply:
In the OTel adapter I do default to internal if no span kind was specified.

I left zero unspecified as I felt the above is implementation specific and should be left to the adapters.  However, if we (the SDK team) are comfortable with stating that unspecified always defaults to internal for any implementation I can change it.

And just for the record, changing the value post release would be a breaking change, so we should be confident in our decision.

Reply:
In fact, having the default be impl-specific means I should also remove the "this is the default" comment.

Reply:
If there are no objections, I'll leave this as-is and remove the comment about the default.

Reply:
sounds good to me

Reply:
Updated comment